### PR TITLE
Randomize the location of all heaps

### PR DESCRIPTION
**Pull request by MikeIsAStar**

### CODE REVIEW

File: payload/sp/Payload.cc [hunk 1]
```diff
@@ -66,10 +67,14 @@ static void Init() {
...
     Console::Print("\n");
 
+    Console::Print("Applying security patches...");
 #ifndef GDB_COMPATIBLE
     OSDisableCodeExecOnMEM1Hi16MB();
 #endif
 
+    Heap_RandomizeMEM1Heaps();
+    Heap_RandomizeMEM2Heaps();
+
```
Comment (stblr) [inline payload/sp/Payload.cc:77]: Maybe move it after the other `OSAlloc` calls.

File: payload/sp/security/Heap.c [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#include "Heap.h"
+
+#include <revolution.h>
+#include <vendor/libhydrogen/hydrogen.h>
+
+#define RANDOMIZATION_SPACE_BYTES (1024 * 8)
+
+void Heap_RandomizeMEM1Heaps(void) {
+    OSAllocFromMEM1ArenaLo(hydro_random_uniform(RANDOMIZATION_SPACE_BYTES), 32);
+    OSAllocFromMEM1ArenaHi(hydro_random_uniform(RANDOMIZATION_SPACE_BYTES), 32);
```
Comment (stblr) [inline payload/sp/security/Heap.c:10]: As discussed, the high size should probably be `RANDOMIZATION_SPACE_BYTES` minus the low size.

**Review by stblr — CHANGES_REQUESTED** [review]

File: payload/sp/Payload.cc (hunk 1 above)
Comment (MikeIsAStar) [reply payload/sp/Payload.cc]: Fixed in commit [76b250e](https://github.com/stblr/mkw-sp/pull/397/commits/76b250ecfa4a82398d2b6d9d54fd06663c1047fa).

File: payload/sp/security/Heap.c (hunk 2 above)
Comment (MikeIsAStar) [reply payload/sp/security/Heap.c]: Fixed in commit [a021de8](https://github.com/stblr/mkw-sp/pull/397/commits/a021de8637f23200a37c53333f2b37692da471da).

**Review by stblr — APPROVED** [review]

**Pull request merged**